### ISSUE TITLE
TechDocs: Improve GCS credentials handling

### ISSUE THREAD

**Issue by ayshiff** [backstage/backstage#3947]
## Feature Suggestion

Same logic as for S3 #3946

We should be able to load GCS credentials from the environment in the [GCS publisher](https://github.com/backstage/backstage/blob/master/packages/techdocs-common/src/stages/publish/googleStorage.ts).

## Possible Implementation

Handle GCS configuration apart from explicitly defining configs.

By default, when the SDK loads, it automatically searches credentials in the environment.

So we can instantiate the SDK without explicitly defining credentials:

```ts
const storage = new Storage();
```

To recap, we should have the following workflow:

<p align="center">
<img src="https://user-images.githubusercontent.com/32459935/103825916-0975a500-5076-11eb-80b0-d31aac349709.png" width="600" />
</p>

## Context

PR which added Google Cloud Storage support - #3432
